parsl/executors/taskvine/exec_parsl_function.py

```diff
@@ -115,26 +113,23 @@ def encode_byte_code_function(user_namespace, fn, fn_name, args_name, kwargs_nam
 def load_function(map_file, function_file, argument_file):
     # Decodes the function and its file arguments to be executed into
     # function_code, and updates a user namespace with the function name and
-    # the variable named result_name. When the function is executed, its result
+    # the variable named `result_name`. When the function is executed, its result
     # will be stored in this variable in the user namespace.
     # Returns (namespace, function_code, result_name)
 
-    # Create the namespace to isolate the function execution.
-    user_ns = locals()
-    user_ns.update({'__builtins__': __builtins__})
-
-    packed_function = load_pickled_file(function_file)
-    packed_argument = load_pickled_file(argument_file)
-
-    fn = unpack_object(packed_function, user_ns)
-    args_dict = unpack_object(packed_argument, user_ns)
+    fn = unpack_object_from_file(function_file)
+    args_dict = unpack_object_from_file(argument_file)
     fn_args = args_dict['args']
     fn_kwargs = args_dict['kwargs']
     fn_name = 'parsl_tmp_func_name'
 
-    mapping = load_pickled_file(map_file)
+    mapping = unpack_object_from_file(map_file)
     remap_all_files(mapping, fn_args, fn_kwargs)
 
+    # Create the namespace to isolate the function execution.
```

Comment:
this block isn't creating a new namespace, I think? it's using and updating the local variables namespace of the current function invocation (if I correctly understand what `locals()` returns).

This is something htex and the workqueue executor do as well, and I wonder if they should all actually be starting with an empty dictionary, `{}`, instead of `locals()`.